examples/cis2-multi-royalties/src/lib.rs

```diff
@@ -0,0 +1,1468 @@
+        owner: &Address,
+        state_builder: &mut StateBuilder<S>,
+        royalty: u8,
+    ) {
+        self.tokens.insert(*token_id);
+        self.token_details.insert(*token_id, TokenDetails {
+            minter: *owner,
+            royalty,
+        });
+
+        let mut owner_state =
+            self.state.entry(*owner).or_insert_with(|| AddressState::empty(state_builder));
+        let mut owner_balance = owner_state.balances.entry(*token_id).or_insert(0.into());
+        *owner_balance += amount;
+    }
+
+    /// Check that the token ID currently exists in this contract.
+    #[inline(always)]
+    fn contains_token(&self, token_id: &ContractTokenId) -> bool { self.tokens.contains(token_id) }
+
+    /// Get the current balance of a given token id for a given address.
+    /// Results in an error if the token id does not exist in the state.
+    fn balance(
+        &self,
+        token_id: &ContractTokenId,
+        address: &Address,
+    ) -> ContractResult<ContractTokenAmount> {
+        ensure!(self.contains_token(token_id), ContractError::InvalidTokenId);
+        let balance = self.state.get(address).map_or(0.into(), |address_state| {
+            address_state.balances.get(token_id).map_or(0.into(), |x| *x)
+        });
+        Ok(balance)
+    }
+
+    /// Check if an address is an operator of a given owner address.
+    fn is_operator(&self, address: &Address, owner: &Address) -> bool {
+        self.state
+            .get(owner)
+            .map(|address_state| address_state.operators.contains(address))
+            .unwrap_or(false)
+    }
+
+    /// Update the state with a transfer.
+    /// Results in an error if the token id does not exist in the state or if
+    /// the from address have insufficient tokens to do the transfer.
+    fn transfer(
+        &mut self,
+        token_id: &ContractTokenId,
+        mut amount: ContractTokenAmount,
+        from: &Address,
+        to: &Address,
+        state_builder: &mut StateBuilder<S>,
+        royalties: &Option<CheckRoyaltyResult>,
+    ) -> ContractResult<()> {
+        ensure!(self.contains_token(token_id), ContractError::InvalidTokenId);
+        // A zero transfer does not modify the state.
+        if amount == 0.into() {
+            return Ok(());
+        }
+        // Get the `from` state and balance, if not present it will fail since the
+        // balance is interpreted as 0 and the transfer amount must be more than
+        // 0 as this point.;
+        {
+            let mut from_address_state =
+                self.state.entry(*from).occupied_or(ContractError::InsufficientFunds)?;
+            let mut from_balance = from_address_state
+                .balances
+                .entry(*token_id)
+                .occupied_or(ContractError::InsufficientFunds)?;
+            ensure!(*from_balance >= amount, ContractError::InsufficientFunds);
+            *from_balance -= amount;
+        }
+
+        if royalties.is_some() {
```

Comment:
You can simplify the syntax to remove the `unwrap()` similar to:

https://github.com/Concordium/concordium-rust-smart-contracts/pull/281#discussion_r1224219364